### PR TITLE
New Mutation Category: Lamia

### PR DESCRIPTION
**Pull request by VMSolidus**
Alright, this has actually been asked of me to do for awhile, both by my friends and several people on the forums awhile back. With this PR, I hope to add the ability for survivors to become a snek girl. For now I only have a single 'style' of snake tail that can be obtained, and that's an armoured lamia. Future PRs will include the ability to become a water snake, constrictor, or a python. 

This is also now the first entirely new mutation category, since mutations got jsonized!

The Lamia Mutation tree focuses mostly on mixing some of the benefits and downsides of lizard and elf. A pre-threshhold lamia is vulnerable, but still has access to a few beneficial traits to make up for it, while also having a big speed debuff. Post threshhold lamia becomes a sneeky snake worthy of being called SNAKE.

### CODE REVIEW

**Comment by evilexecutive** [comment]
And now for the slow crawl of "Wait for the interpreter to report an error, so I can fix it"...

**Comment by cainiaowu** [comment]
Or you can try to load them locally since it only has json change.

**Comment by evilexecutive** [comment]
I'm retarded like that, and only slightly off kilter from some wine. For now I just want to relax and wonder why a perfectly ordinary line is showing up red on the editor, despite having no conceivable problem to me.

**Comment by evilexecutive** [comment]
Why, how, and wut. It wants something different from a { to start the set, but I don't have a freakin' clue what's wrong now. >.<

**Comment by DangerNoodle** [comment]
Boop all the snoots?

**Comment by evilexecutive** [comment]
Yes, all of the snoots must be booped. And I think I'm done for the night. Tomorrow I'm adding in snakeskin on a separate PR, if I can figure out how that code works.

**Comment by chaosvolt** [comment]
[Vore jokes intensify]

File: data/json/mutations.json [hunk 1]
```diff
@@ -1423,6 +1423,76 @@
...
+    },{
+        "type" : "mutation",
+        "id" : "LAMIA_TAIL_JEUVENILE",
+        "name" : "Short Snake Tail",
+        "points" : 14,
+        "visibility" : 15,
+        "ugliness" : 6,
+        "description" : "Your legs have fused together to form the beginings of a snake tail. While inherently useless now, you hope it will grow much bigger",
+        "cancels" : ["SCALES", "FEATHERS", "CHITIN", "PLANTSKIN", "FELINE_FUR", "LUPINE_FUR", "CHITIN_FUR"],
+        "prereqs" : ["LAMIA_TAIL_JEUVENILE"],
```
Comment (Coolthulhu) [inline data/json/mutations.json]: Requirement of itself?

File: data/json/mutations.json [hunk 2]
```diff
@@ -1423,6 +1423,76 @@
...
+        "threshreq" : ["THRESH_LAMIA"],
+        "category" : ["MUTCAT_LAMIA"],
+        "purifiable" : false,
+        "restricts_gear" : [ "FOOT_L", "FOOT_R", "LEG_L", "LEG_R"],
+        "passive_mods" : {
+            "dex_mod" : -2,
+            "encumbrance_always" : [
```
Comment (Coolthulhu) [inline data/json/mutations.json]: Encumbrance and passive mods are two separate json fields, they must not be nested within each other.

**Comment by Coolthulhu** [comment]
Can we keep the memes lower? I mean in the in-game descriptions, comments are fine.

File: data/json/mutations.json [hunk 3]
```diff
@@ -1423,6 +1423,76 @@
...
+        "category" : ["MUTCAT_LAMIA"]
+    },{
+        "type" : "mutation",
+        "id" : "LAMIA_TAIL",
+        "name" : "Snake Tail",
+        "points" : 14,
+        "visibility" : 15,
+        "ugliness" : 6,
+        "description" : "Your legs now terminate in an enormous noodly tail. You can now move greatly faster.",
+        "cancels" : ["SCALES", "FEATHERS", "CHITIN", "PLANTSKIN", "FELINE_FUR", "LUPINE_FUR", "CHITIN_FUR"],
```
Comment (Coolthulhu) [inline data/json/mutations.json]: Should probably cancel foot and leg mutations. Otherwise you could get tail+tentacle legs or tail+padded feet.

File: data/json/mutations.json [hunk 4]
```diff
@@ -1423,6 +1423,76 @@
...
+            "dex_mod" : 2
+        }
+    },{
+        "type" : "mutation",
+        "id" : "LAMIA_TAIL_JEUVENILE",
+        "name" : "Short Snake Tail",
+        "points" : 14,
+        "visibility" : 15,
+        "ugliness" : 6,
+        "description" : "Your legs have fused together to form the beginings of a snake tail. While inherently useless now, you hope it will grow much bigger",
```
Comment (Coolthulhu) [inline data/json/mutations.json]: Purely useless mutations that lead to good ones aren't a good idea, they just end up forcing you to consume 2-3 mutagens in a row instead of one.
They aren't realistic either, because current mutations all happen the moment you gain them, so there is no growing process.

It would be better to give it some advantages of the upgraded trait (stat boosts), but with some disadvantages (leg/foot encumbrance).

File: data/json/mutations.json [hunk 5]
```diff
@@ -1423,6 +1423,76 @@
...
+        "prereqs" : ["LAMIA_TAIL"],
+        "threshreq" : ["THRESH_LAMIA"],
+        "category" : ["MUTCAT_LAMIA"],
+        "armor" : [ 
+            { "parts" : "LEG_L", "bash" : 11, "cut" : 24 }, 
+            { "parts" : "LEG_R", "bash" : 11, "cut" : 24 },
+            { "parts" : "FOOT_L", "bash" : 11 , "cut" : 24 },
+            { "parts" : "FOOT_R", "bash" : 11 , "cut" : 24 },
+            { "parts" : "MOUTH", "cut" : 14}
+            ]
```
Comment (Coolthulhu) [inline data/json/mutations.json]: A minor acid protection (2-3 points) on the tail would be very useful, since you can't wear rubber boots.

**Comment by Coolthulhu** [comment]
Not too sure about the tail part. You'll still have 2 legs and 2 feet in the hp display, will still be able to break them and be forced to wear a splint over them etc.

**Comment by DangerNoodle** [comment]
That might require ability for mutations to change body part definitions.

Then again, wings and tails do not show as valid targets.

**Comment by Coolthulhu** [comment]
Wings and tails are new body parts and losing them wouldn't impair the survivor much.
Legs and feet merging into one but still allowing boots (`"OVERSIZE"` gear skips gear restriction), being damaged separately etc.

**Comment by evilexecutive** [comment]
Having a Lamia Tail is NOT intended to make you immune to having your "legs" broken. It's still a complicated structure with muscles and bones that can still break. A snake can actually break the bones on one half of its tail, and get hobbled in such a way it struggles to move in one direction.

File: data/json/mutations.json (hunk 5 above)
Comment (evilexecutive) [inline data/json/mutations.json]: I'll add this in to the standard scales as their one token benefit.

File: data/json/mutations.json (hunk 4 above)
Comment (evilexecutive) [inline data/json/mutations.json]: There is in this case. Taking a lamia mutagen through the IV forces you to sleep for awhile, so that you can process the much larger physiology changes. I mean you literally end up with a tail thats somewhere around 15 to 25 feet. And that's reflected in the mutagens side of this PR.

**Comment by evilexecutive** [comment]
Yea that's a bit silly to wear boots on your tail. Although I have no idea how to restrict oversized gear as well.

File: data/json/mutations.json [hunk 6]
```diff
@@ -1447,9 +1447,14 @@
         "points" : 4,
         "visibility" : 10,
         "ugliness" : 1,
-        "description" : "You have grown a smooth, scale-like skin over your legs and mouth. While luxuriously soft, they offer no real benefit",
+        "description" : "You have grown a smooth, scale-like skin over your legs and mouth. While luxuriously soft, they also give some small resistance to acid",
         "cancels" : ["SCALES", "FEATHERS", "CHITIN", "PLANTSKIN", "FELINE_FUR", "LUPINE_FUR", "CHITIN_FUR"],
         "category" : ["MUTCAT_LAMIA"]
```
Comment (Zireael07) [inline data/json/mutations.json]: Missing comma!

Comment (evilexecutive) [inline data/json/mutations.json]: Got it!

File: data/json/items/comestibles/mutagen.json [hunk 7]
```diff
@@ -54,6 +54,15 @@
         "flags" : "MUTAGEN_BIRD"
     },
     {
+        "id" : "iv_mutagen_lamia",
+        "copy-from" : "iv_mutagen_flavor",
+        "type" : "COMESTIBLE",
+        "name" : "lamia serum",
+        "description" : "A prismatic mutagen with a scaly texture. You need a syringe to inject it... if you really want to?",
```
Comment (Rivet-the-Zombie) [inline data/json/items/comestibles/mutagen.json]: Double space after all sentence breaks, please.

File: data/json/mutations.json [hunk 8]
```diff
@@ -1423,6 +1423,82 @@
...
+            [ "FOOT_L", 15 ],
+            [ "FOOT_R", 15 ]]
+    },{
+        "type" : "mutation",
+        "id" : "LAMIA_SCALES",
+        "name" : "Lamia Scales",
+        "points" : 4,
+        "visibility" : 10,
+        "ugliness" : 1,
+        "description" : "You have grown a smooth, scale-like skin over your legs and mouth. While luxuriously soft, they also give some small resistance to acid",
```
Comment (Rivet-the-Zombie) [inline data/json/mutations.json]: Skin grown over the mouth? Do they still have mouths after this? This description's wording is a bit confusing.

Comment (evilexecutive) [inline data/json/mutations.json]: Yea, I meant to say cheeks. My reference image for this whole spiel is obviously miia, and that lamia had snake scales on her cheeks.

**Comment by evilexecutive** [comment]
I think at this point all of the errors are fixed, and this PR is ready to be pushed to mainline. Thanks everyone for helping me out with my work, and for pointing out all the errors and mistakes!

File: data/json/mutations.json (hunk 8 above)
Comment (Rivet-the-Zombie) [inline data/json/mutations.json]: I have no idea who that is, but thank you for changing it.

Comment (evilexecutive) [inline data/json/mutations.json]: Oh dear god you poor soul, you have never seen Monster Musume have you? Miia was a lamia from it, and one of the main characters from said show. It's an echi series, and a ridiculously popular one at that.

Comment (Rivet-the-Zombie) [inline data/json/mutations.json]: It sounds like I'm not the right demographic for it, plus I don't do Japanese cartoons; regardless, I like the sound of this strange new MUTCAT.

File: data/json/mutations.json [hunk 9]
```diff
@@ -1423,6 +1423,82 @@
...
+        "changes_to" : ["LAMIA_TAIL"],
+        "purifiable" : true,
+        "restricts_gear" : [ "FOOT_L", "FOOT_R", "LEG_L", "LEG_R"],
+        "passive_mods" : {
+            "dex_mod" : -2},
+        "encumbrance_always" : [
+            [ "LEG_L", 50 ],
+            [ "LEG_R", 50 ],
+            [ "FOOT_L", 50 ],
+            [ "FOOT_R", 50 ] ]
```
Comment (Coolthulhu) [inline data/json/mutations.json]: Here and in every similar example above:
Bracket should be closed on a new line, unless the whole block is on one line.
For example:
```
"one_line" : [ "ONE", "LINE" ],
"many_lines" : [
    "MANY",
    "LINES"
]

File: data/json/recipes/recipe_medsandchemicals.json [hunk 10]
```diff
@@ -798,6 +798,68 @@
...
   "type" : "recipe",
+  "result": "mutagen_lamia",
+  "category": "CC_CHEM",
+  "subcategory": "CSC_CHEM_MUTAGEN",
+  "skill_used": "cooking",
+  "skills_required": [ "firstaid", 1 ],
+  "difficulty": 9,
+  "time": 10000,
+  "reversible": false,
+  "autolearn": false,
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_medsandchemicals.json]: Reversible and autolearn being set to false is redundant - those two lines can be deleted

File: data/json/recipes/recipe_medsandchemicals.json [hunk 11]
```diff
@@ -798,6 +798,68 @@
...
+  "book_learn": [[ "recipe_maiar", 8 ]],
+  "qualities":[
+    {"id":"CHEM","level":2,"amount":1}
+  ], "tools": [
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_medsandchemicals.json]: `"amount":1` is redundant.
The line above can be condensed into one line (ie. qualities and the one quality in it). Otherwise "tools" should be on a new line.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 12]
```diff
@@ -798,6 +798,68 @@
...
+  ],
+  "components": [
+    [
+      [ "mutagen_slime", 1 ],
+      [ "mutagen_lizard", 1],
+      [ "mutagen_bird", 1]
+    ],
+    [
+      [ "cooking_oil", 1 ],
+      [ "chitin_piece", 4 ]
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_medsandchemicals.json]: Cooking oil is much cheaper than chitin. Don't you mean chitin meal or something like that?
Also, cooking oil is vegetable matter.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 10 above)
Comment (evilexecutive) [inline data/json/recipes/recipe_medsandchemicals.json]: If it's redundant here, it's redundant across the entirety of all of the mutation recipes. Should I then remove them from all of the other recipes while I'm at it, if that'll optimize the code a little more?

File: data/json/recipes/recipe_medsandchemicals.json (hunk 12 above)
Comment (evilexecutive) [inline data/json/recipes/recipe_medsandchemicals.json]: No, I have cooking oil as a placeholder (and a cheeky reference to snake oil)there for a future PR, where I add in Snakeskin and other snakeskin related items. I'm not going to address it here, because I haven't set-out to make that kind of stuff in this PR.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 10 above)
Comment (Coolthulhu) [inline data/json/recipes/recipe_medsandchemicals.json]: Mass fixes should be their own PRs. Just keep this one up to standards.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 12 above)
Comment (Coolthulhu) [inline data/json/recipes/recipe_medsandchemicals.json]: Don't add placeholders like that. It's trivial to add it later and now it only causes extra "wtf"s.

File: data/json/recipes/recipe_medsandchemicals.json (hunk 10 above)
Comment (evilexecutive) [inline data/json/recipes/recipe_medsandchemicals.json]: Ah shoot, I went through and fixed the entirety of it after leaving my comment, then you commented while I went ahead and made said fixes. >.<

File: data/json/recipes/recipe_medsandchemicals.json (hunk 12 above)
Comment (evilexecutive) [inline data/json/recipes/recipe_medsandchemicals.json]: Yea, I've gone ahead and removed it. I'll leave Chitin there though.

**Comment by pisskop** [comment]
Has the endstage been tested?
Do the mutations offered for lamia also contribute to select other categories?

_>.>_
_<.<_

Do I have to be a snek _girl_?

**Comment by evilexecutive** [comment]
Yes you can be a Snek Boy. Endstage is working fine, although there aren't too many lamia mutations post-threshhold at this time(and there won't be until I figure out how to code things like unique attacks from mutations). 

None of the post-thresh lamia mutations contribute to other categories at this time, ESPECIALLY not the tail. I explicitly do not want the tail to be obtainable from other categories because the pre-thresh tail is actually a very strong negative trait. So it would be mighty inconvenient to other mutation trees if you could just spontaneously obtain a 3 foot long snake tail.

**Comment by ABlitzkriegOfButts** [comment]
This seems really kewl, Is there a download link yet?

**Comment by evilexecutive** [comment]
If the gods are willing, this will be mainlined into an experimental branch. So eventually, provided kevin signs off on it.

**Comment by evilexecutive** [comment]
Wait, why is this now producing errors? Also, how did it go so long without ever being pulled that it now starts producing errors?..

This isn't something  I can fix..

**Comment by DangerNoodle** [comment]
Jently has been having issues as of late, it would seem.

**Comment by evilexecutive** [comment]
Well, whatever the problem is, it doesn't seem to be on my end of the code. I hope it gets resolved eventually, and that my PR gets pulled. >.<

**Comment by DangerNoodle** [comment]
Same, there are multiple pull requests this will have an effect on, but I do not know the full extent of the "out of memory" problem at present.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 13]
```diff
@@ -2641,11 +2540,9 @@
   "skills_required": [ "cooking", 1 ],
   "difficulty": 4,
   "time": 20000,
-  "reversible": false,
```
Comment (mugling) [inline data/json/recipes/recipe_medsandchemicals.json]: Try and avoid unrelated changes in the same PR

**Comment by mugling** [comment]
JSON load errors and touches a lot of unrelated data (especially recipes) so will merge conflict easily.

Closing as it's going to merge conflict as I merge other PR's and the fastest route for you would be to submit a new PR containing just the relevant changes and with the JSON errors fixed.

Alternatively you can rebase and reopen this PR if you have the git-fu

File: data/json/mutations.json (hunk 4 above)
Comment (yowshi) [inline data/json/mutations.json]: then the blobs sensory degredation should be removed no?
